### PR TITLE
Move LocationAlerts and PersonAlerts to separate endpoints

### PR DESCRIPTION
**Pull request by LBHCallumM**
[Story](https://www.pivotaltracker.com/story/show/181511813)

## Describe this PR

The endpoint for getting LocationAlerts sometimes returns 504 BadGateway when it receives too many requests.

If this occurs, the properties endpoint will fail to load, preventing users from raising any repairs.

To resolve this issue, I have moved both alerts to separate endpoints. This means, if the API fails, users will still be able to raise a repair. I have separated them into two requests, rather than one, because personAlerts doesn't normally have issues, only the location alerts.

Added New Endpoints:

`/api/v2/properties/tenancyAgreementReference}/person-alerts`
`/api/v2/properties/{propertyReference}/location-alerts`

Response object for Both:
```
{
    "reference": "propRef or tenancyAgreementReferense",
    "alerts": [
        {
            "type": "",
            "comments": "",
            "startDate": "",
            "endDate": ""
        }
        // same as existing format
    ]
}
```

The `TenancyAgreementReference` value will be added to the property response `response.tenure.tenancyAgreementReference`.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

### CODE REVIEW

File: RepairsApi/V2/Controllers/PropertiesController.cs [hunk 1]
```diff
@@ -68,6 +78,34 @@ public async Task<IActionResult> GetProperty([FromRoute] GetPropertyRequest quer
             return Ok(property.ToResponse());
         }
 
+        [HttpGet]
+        [Route("{tenancyAgreementReference}/person-alerts")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(AlertsViewModel), 200)]
```
Comment (Asura5) [inline RepairsApi/V2/Controllers/PropertiesController.cs:84]: Can it also produce a 404 if the person isn't found, or a 400 if the request is malformed?

File: RepairsApi/V2/Controllers/PropertiesController.cs [hunk 2]
```diff
@@ -68,6 +78,34 @@ public async Task<IActionResult> GetProperty([FromRoute] GetPropertyRequest quer
...
+        {
+            var alerts = await _personAlertsUseCase.ExecuteAsync(tenancyAgreementReference);
+
+            return Ok(alerts.ToResponse());
+        }
+
+        [HttpGet]
+        [Route("{propertyReference}/location-alerts")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(AlertsViewModel), 200)]
```
Comment (Asura5) [inline RepairsApi/V2/Controllers/PropertiesController.cs:98]: Can it also produce a 404 if the location isn't found, or a 400 if the request is malformed?

File: RepairsApi/V2/Factories/ResponseFactory.cs [hunk 3]
```diff
@@ -31,6 +31,24 @@ public static CautionaryAlertResponseList ToResponse(this AlertList domain)
             };
         }
 
+        public static AlertsViewModel ToResponse(this PropertyAlertList domain)
```
Comment (Asura5) [inline RepairsApi/V2/Factories/ResponseFactory.cs:34]: SOLID - breaks DRY. Refactor available: have a common superclass for PropertyAlertList & PersonAlertList and use `typeof` to set the refernces

File: RepairsApi/V2/Controllers/PropertiesController.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi/V2/Controllers/PropertiesController.cs:84]: You're right about the 404 response. But neither endpoint has any validation for the propertyReference or tenancyAgreementReference.

File: RepairsApi/V2/Factories/ResponseFactory.cs (hunk 3 above)
Comment (LBHCallumM) [reply RepairsApi/V2/Factories/ResponseFactory.cs]: Also, the ListAlerts endpoint isn't used anymore. I checked cloudwatch for any logs within the last 2 months.

File: RepairsApi/V2/Controllers/PropertiesController.cs (hunk 2 above)
Comment (LBHCallumM) [reply RepairsApi/V2/Controllers/PropertiesController.cs:101]: Discussed not to include 404.

Comment (LBHCallumM) [reply RepairsApi/V2/Controllers/PropertiesController.cs:101]: Added whitespace checks which will returns badRequest.

**Review by pixeltrix — APPROVED** [review]

**Pull request merged**